packages/server/src/sdk/app/rows/search/sqs.ts

```diff
@@ -240,18 +237,19 @@ async function runSqlQuery(
 function resyncDefinitionsRequired(status: number, message: string) {
   // pre data_ prefix on column names, need to resync
   return (
-    (status === 400 && message?.match(USER_COLUMN_PREFIX_REGEX)) ||
-    // default tables aren't included in definition
-    (status === 400 &&
-      DEFAULT_TABLE_IDS.find(tableId => message?.includes(tableId))) ||
+    // there are tables missing - try a resync
+    (status === 400 && message.match(MISSING_TABLE_REGX)) ||
+    // there are columns missing - try a resync
+    (status === 400 && message.match(MISSING_COLUMN_REGEX)) ||
     // no design document found, needs a full sync
     (status === 404 && message?.includes(SQLITE_DESIGN_DOC_ID))
   )
 }
 
 export async function search(
   options: RowSearchParams,
-  table: Table
+  table: Table,
+  opts?: { retries?: number }
```

Comment:
This seems confusing to me, as I would expect this to be the remaining attempts, not the current ones. For this case, either having a private function or just a boolean "retrying" would work